src/server/controllers/templateController.js

```diff
@@ -0,0 +1,102 @@
+const Quiz = require('../models/quiz');
+const QuizTemplate = require('../models/templates');
+
+const fetchTemplates = async (req, res) => {
+  try {
+    const templates = await QuizTemplate.find();
+    return res.json(templates);
+  } catch (error) {
+    console.log(error);
+    return res.status(500).json({ error: 'Failed to fetch templates' });
+  }
+};
+
+const createTemplate = async (req, res) => {
+  try {
+    const userId = req.user.id;
+    const { name, description, questions, duration } = req.body;
+
+    const template = new QuizTemplate({
+      name,
+      description,
+      questions,
+      duration,
+      createdBy: userId,
+    });
+
+    await template.save();
+    return res
+      .status(201)
+      .json({ message: 'Template created successfully', template });
+  } catch (error) {
+    console.log(error);
+    return res
+      .status(500)
+      .json({ error: 'Server error while creating template' });
+  }
+};
+
+const cloneTemplate = async (req, res) => {
+  try {
+    const userId = req.user.id;
+    const template = await QuizTemplate.findById(req.params.id);
+
+    if (!template) {
+      return res.status(404).json({ error: 'Template not found' });
+    }
+
+    const newQuiz = new Quiz({
+      title: template.name,
+      questions: template.questions,
+      duration: template.duration,
+      author: userId,
+    });
+
+    await newQuiz.save();
+    return res
+      .status(201)
+      .json({ message: 'Quiz created from template', quiz: newQuiz });
+  } catch (error) {
+    return res
+      .status(500)
+      .json({ error: 'Failed to create quiz from template' });
+  }
+};
+
+const deleteTemplate = async (req, res) => {
+  try {
+    const userId = req.user.id;
+    const templateId = req.params.id;
+
+    console.log('Template ID:', templateId);
+
+    const template = await QuizTemplate.findById(templateId);
+    if (!template) {
+      return res.status(404).json({ error: 'Template not found' });
+    }
+
+    if (String(template.createdBy) !== String(userId)) {
```

Comment:
Wouldn't it be better to keep template application specific rather than user specific. So anyone with the admin role can delete the template as well as create one .

Reply:
On it!